dipy/testing/tests/test_decorators.py

```diff
@@ -51,3 +51,36 @@ def f():
     del HAVE_AMODULE
     f.__doc__ = docstring
     assert_raises(NameError, doctest_skip_parser, f)
+
+def test_warn_on_positional_kwonly():
```

Comment:
Maybe it would be easier to test this concept as @WassCodeur did in [his implementation](https://github.com/fury-gl/fury/pull/888) of this same decorator for Fury. He defines only 1 function and tests it in all possible cases, instead of two functions. I think it is simpler.